tests/migrations/test_state.py

```diff
@@ -1016,6 +1028,71 @@ class Meta:
         choices_field = Author._meta.get_field('choice')
         self.assertEqual(list(choices_field.choices), choices)
 
+    def test_relations_add_model(self):
+        project_state = ProjectState()
+        self.set_up_test_model('tests', project_state)
+        self.assertEqual(set(project_state.relations['tests', 'user']), {('tests', 'comment'), ('tests', 'blog')})
+        self.assertEqual(set(project_state.relations['tests', 'self_model']), {('tests', 'self_model')})
```

Comment:
Do we need to call `set()`? is seems unnecessary, see my previous https://github.com/django/django/pull/14587#discussion_r692706334. Please check all new tests.

If the order is not important you can use `assertCountEqual()` instead of `assertEqual()`.

Reply:
without `set()` it will return an object of `defaultdict`. That's why I found it necessary.

Reply:
Though the order is not important, still we can not use `assertCountEqual()` as the name is important and is to be verified.